### PR TITLE
[RDBMS] Update Postgres restore command to support available zone

### PR DESCRIPTION
**Pull request by DaeunYim**
**Description**

- Users can choose available zone when restoring the PG server.
- The default value for restore time is set to current time. 
- Added password test
- [IMPORTANT] Added Storage service endpoint to vnet/subnet when provisioning a database server with that to address OPAPH issue (restore failure in PROD). 

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
RDBMS

**Comment by yonzhan** [comment]
RDBMS

**Comment by DaeunYim** [comment]
Hi! This change must be checked in to prevent PostgreSQL restore failure caused by WAL replica storage issue. I would very much appreciate if you could review this before the code complete day!

**Comment by yonzhan** [comment]
@evelyn-ys will help with this PR.

File: src/azure-cli/azure/cli/command_modules/rdbms/_params.py [hunk 1]
```diff
@@ -97,7 +98,7 @@ def _complex_params(command_group):
         with self.argument_context('{} server restore'. format(command_group)) as c:
             c.argument('server_name', options_list=['--name', '-n'], arg_type=overriding_none_arg_type)
             c.argument('source_server', options_list=['--source-server', '-s'], help='The name or resource ID of the source server to restore from.')
-            c.argument('restore_point_in_time', options_list=['--restore-point-in-time', '--pitr-time'], help='The point in time to restore from (ISO8601 format), e.g., 2017-04-26T02:10:00+08:00')
+            c.argument('restore_point_in_time', options_list=['--restore-point-in-time', '--pitr-time'], help='The point in time in UTC to restore from (ISO8601 format), e.g., 2017-04-26T02:10:00+08:00')
```
Comment (evelyn-ys) [inline src/azure-cli/azure/cli/command_modules/rdbms/_params.py:101]: Why no `default` and no `zone` here? It looks different from `'{} flexible-server restore'.format(command_group)`

**Review by evelyn-ys — CHANGES_REQUESTED** [review]
Can you add tests for:
- Users can choose available zone when restoring the PG server.
- The default value for restore time is set to current time.
- [IMPORTANT] Added Storage service endpoint to vnet/subnet when provisioning a database server with that to address OPAPH issue (restore failure in PROD).

File: src/azure-cli/azure/cli/command_modules/rdbms/_params.py (hunk 1 above)
Comment (DaeunYim) [reply src/azure-cli/azure/cli/command_modules/rdbms/_params.py:101]: The PG flexible server team reached out to me to add the zone parameter for flexible server to align it with the portal experience.

**Comment by DaeunYim** [comment]
> Can you add tests for:
> 
> * Users can choose available zone when restoring the PG server.
> * The default value for restore time is set to current time.
> * [IMPORTANT] Added Storage service endpoint to vnet/subnet when provisioning a database server with that to address OPAPH issue (restore failure in PROD).

Added test for available zone, I think the other two are hard to add tests for..

**Review by evelyn-ys — APPROVED** [review]

**Pull request merged**